### PR TITLE
fix: Map language types to OpenAPI types

### LINKED ISSUE #445: openapi: openapi does not generate valid parameter types
The docs are generated with language-native parameter type names instead of openapi-specced names. Examples:

generated from Python AppMaps:
```yaml
        - name: secret
          in: path
          schema:
            type: builtins.str
          required: true
        - name: thumbnail
          in: query
          schema:
            type: builtins.bool
```

generated from Java AppMaps:
```yaml
        - name: id
          in: path
          schema:
            type: java.lang.string
          required: true
```

Valid types for these two examples are `boolean`, `string`.

### PR DESCRIPTION
**Pull request by apotterri**
Add type mappings for Python and Java. This ensures that all types that
can appear in AppMaps will get mapped to the corresponding OpenAPI type.

Fixes #445.

### CODE REVIEW

File: packages/cli/src/openapi/util.js [hunk 1]
```diff
@@ -1,24 +1,60 @@
...
+      case 'java.lang.string':
+        return 'string';
+      default:
+        return undefined;
+    }
+  };
+
+  const mapper = (t) => mapRubyType(t) || mapPythonType(t) || mapJavaType(t);
+
+  return mapper(className.toLowerCase()) || className;
```
Comment (apotterri) [inline packages/cli/src/openapi/util.js:57]: @kgilpin Is there a reason to ignore case when comparing types here?

Comment (apotterri) [inline packages/cli/src/openapi/util.js:57]: @kgilpin @ptrdvrk The previous version of this code also just returns the class name if no other mapping is found. But, is this really the behavior we want? At the very least, it seems like we should also log a warning that we're generating an invalid spec.

**Review by ptrdvrk — APPROVED** [review]
@apotterri from a user's perspective this look good to me, thank you. swagger.io still complains about one issue in the Misago docs - but that is a different problem.

https://app.swaggerhub.com/apis/ptrdvrk8/misago/v1

**Comment by apotterri** [comment]
Thanks, @ptrdvrk. Do you have an opinion about what the `type` of a parameter should be set to for unmapped types, or if we should log a message if we see one? For example, what should we have done when we encountered `java.lang.String`?

**Comment by ptrdvrk** [comment]
@apotterri I would log a warning but default to the class name, with a comment in the generated doc file. I think it is more valuable to have an openapi document that requires a few edits vs not having one at all.

**Comment by kgilpin** [comment]
Me too.

**Comment by apotterri** [comment]
@ptrdvrk Logging the warning is easy. Getting a comment into the generated doc is going to take a fair bit of work -- the point where we detect the problem is pretty far from where the doc gets created.

Do you want to wait for the comment to be added, or should we do that as a follow on change?

**Comment by kgilpin** [comment]
We don't need the comment at this time. It's going to be fairly clear to
the user what needs to be done.

**Comment by ptrdvrk** [comment]
+1, let's release without comments now.

**Pull request merged**